rir/src/interpreter/interp.cpp

```diff
@@ -22,34 +22,47 @@ extern Rboolean R_Visible;
 
 using namespace rir;
 
-struct CallContext;
-struct FinalCallContext;
-struct FinalDispatchContext;
+struct CallsiteInfo {
```

Comment:
The name is a bit off I feel, since this struct contains a `CallSite*`. Maybe just `CallInfo` or `CallContext` or `DispatchInfo` or similar?
But this is fine too.